### PR TITLE
aws: fix test flake when no IMDS can be found

### PR DESCRIPTION
**Pull request by nbaws**
Commit Message: aws: fix test flake when no IMDS can be found
Additional Description: Lack of IMDS (169.254.169.254 address) can cause a race condition and crash during testing due to the shutdown of cluster manager. This scenario should not occur normally, as cluster manager will still exist and the lack of IMDS is handled gracefully.
Risk Level: Low
Testing: N/A
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by karanvasnani — APPROVED** [review]

File: source/extensions/common/aws/metadata_fetcher.cc [hunk 1]
```diff
@@ -52,6 +52,12 @@ class MetadataFetcherImpl : public MetadataFetcher,
     ASSERT(!request_);
     complete_ = false;
     receiver_ = makeOptRef(receiver);
+
+    // Stop processing if we are shutting down
+    if (cm_.isShutdown()) {
```
Comment (KBaichoo) [inline source/extensions/common/aws/metadata_fetcher.cc:57]: can we have a test for this?

**Review by KBaichoo — COMMENTED** [review]
Just to make sure I understand, you had a manual test where this crashed?

There is no CI flaky test that this was tripping?

/wait

**Comment by nbaws** [comment]
> Just to make sure I understand, you had a manual test where this crashed?
> 
> There is no CI flaky test that this was tripping?
> 
> /wait

The test case would crash when 169.254.169.254 was not available on the host. Unfortunately, where the code was written and also the CI system has this available, leading to the test success. This crash occurs during envoy shutdown in the test case where there is still a request pending, which is now handled by this fix.

I'll write a test case to exercise this scenario.

File: source/extensions/common/aws/metadata_fetcher.cc (hunk 1 above)
Comment (nbaws) [reply source/extensions/common/aws/metadata_fetcher.cc:57]: test case emulates shutdown, as well as the nullptr that would have crashed it had the fix not been in place

**Review by KBaichoo — APPROVED** [review]
Thank you.

**Pull request merged**